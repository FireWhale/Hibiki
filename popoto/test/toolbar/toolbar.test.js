import * as d3 from 'd3';
import tools from "../../src/tools/tools"
import toolBar from "../../src/toolbar/toolbar"

describe("render", function () {
    beforeEach(() => {
        tools.CENTER_GRAPH = false;
        tools.RESET_GRAPH = false;
        tools.SAVE_GRAPH = false;
        tools.TOGGLE_TAXONOMY = false;
        tools.TOGGLE_FULL_SCREEN = false;
        tools.TOGGLE_VIEW_RELATION = false;
        tools.TOGGLE_FIT_TEXT = false;
        document.body.innerHTML = "";
    });

    test('Should render correctly empty', () => {
        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Should render correctly CENTER_GRAPH', () => {
        tools.CENTER_GRAPH = true;

        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Should render correctly RESET_GRAPH', () => {
        tools.RESET_GRAPH = true;

        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Should render correctly SAVE_GRAPH', () => {
        tools.SAVE_GRAPH = true;

        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Should render correctly TOGGLE_TAXONOMY', () => {
        tools.TOGGLE_TAXONOMY = true;

        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Should render correctly TOGGLE_FULL_SCREEN', () => {
        tools.TOGGLE_FULL_SCREEN = true;

        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Should render correctly TOGGLE_VIEW_RELATION', () => {
        tools.TOGGLE_VIEW_RELATION = true;

        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Should render correctly TOGGLE_FIT_TEXT', () => {
        tools.TOGGLE_FIT_TEXT = true;

        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Should render correctly all', () => {
        tools.CENTER_GRAPH = true;
        tools.RESET_GRAPH = true;
        tools.SAVE_GRAPH = true;
        tools.TOGGLE_TAXONOMY = true;
        tools.TOGGLE_FULL_SCREEN = true;
        tools.TOGGLE_VIEW_RELATION = true;
        tools.TOGGLE_FIT_TEXT = true;

        toolBar.render(d3.select("body"));
        expect(document.body.innerHTML).toMatchSnapshot();
    });

});