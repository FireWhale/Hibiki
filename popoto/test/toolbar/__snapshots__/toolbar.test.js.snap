// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Should render correctly CENTER_GRAPH 1`] = `"<div class=\\"ppt-toolbar\\"><span id=\\"popoto-center-menu\\" class=\\"ppt-icon ppt-menu center\\"></span></div>"`;

exports[`render Should render correctly RESET_GRAPH 1`] = `"<div class=\\"ppt-toolbar\\"><span id=\\"popoto-reset-menu\\" class=\\"ppt-icon ppt-menu reset\\"></span></div>"`;

exports[`render Should render correctly SAVE_GRAPH 1`] = `"<div class=\\"ppt-toolbar\\"><span id=\\"popoto-save-menu\\" class=\\"ppt-icon ppt-menu save\\"></span></div>"`;

exports[`render Should render correctly TOGGLE_FIT_TEXT 1`] = `"<div class=\\"ppt-toolbar\\"><span id=\\"popoto-fit-text-menu\\" class=\\"ppt-icon ppt-menu fit-text\\"></span></div>"`;

exports[`render Should render correctly TOGGLE_FULL_SCREEN 1`] = `"<div class=\\"ppt-toolbar\\"><span id=\\"popoto-fullscreen-menu\\" class=\\"ppt-icon ppt-menu fullscreen\\"></span></div>"`;

exports[`render Should render correctly TOGGLE_TAXONOMY 1`] = `"<div class=\\"ppt-toolbar\\"><span id=\\"popoto-taxonomy-menu\\" class=\\"ppt-icon ppt-menu taxonomy\\"></span></div>"`;

exports[`render Should render correctly TOGGLE_VIEW_RELATION 1`] = `"<div class=\\"ppt-toolbar\\"><span id=\\"popoto-toggle-relation\\" class=\\"ppt-icon ppt-menu relation\\"></span></div>"`;

exports[`render Should render correctly all 1`] = `"<div class=\\"ppt-toolbar\\"><span id=\\"popoto-toggle-relation\\" class=\\"ppt-icon ppt-menu relation\\"></span><span id=\\"popoto-reset-menu\\" class=\\"ppt-icon ppt-menu reset\\"></span><span id=\\"popoto-taxonomy-menu\\" class=\\"ppt-icon ppt-menu taxonomy\\"></span><span id=\\"popoto-center-menu\\" class=\\"ppt-icon ppt-menu center\\"></span><span id=\\"popoto-fullscreen-menu\\" class=\\"ppt-icon ppt-menu fullscreen\\"></span><span id=\\"popoto-save-menu\\" class=\\"ppt-icon ppt-menu save\\"></span><span id=\\"popoto-fit-text-menu\\" class=\\"ppt-icon ppt-menu fit-text\\"></span></div>"`;

exports[`render Should render correctly empty 1`] = `"<div class=\\"ppt-toolbar\\"></div>"`;
